### DISCUSSION TITLE
EXTRA_JAVA_OPTIONS="Xmx****m" not working on docker-compose.yml

### DISCUSSION THREAD

**Discussion by marcoamujica** [DependencyTrack/dependency-track#2257] [Q&A] [answered]
Hello, all.  I'm trying to increase my API container's JVM max heap size.  By default, it looks like it's a quarter of what the container has allocated, but I'd like to increase it without having to add much more memory to the container.  I see that there's an EXTRA_JAVA_OPTIONS flag in `/src/main/docker/docker-compose.yml` that should allow me to make changes to the API server's JVM configs.  I added the line to my own YAML just above `deploy:`, but nothing seemed to change.  This is what the line looks like:
- EXTRA_JAVA_OPTIONS="-Xmx6144m"
When I re-run compose, I don't get any errors, but when I go into the container, my max heap size still shows the same.  I was wondering what I was doing wrong.  Thanks in advance.

**Comment by nscuro** [comment, marked as answer]
Hey @marcoamujica, `EXTRA_JAVA_OPTIONS` was only added in 4.7.0 (to be released) and is not available in 4.6.x. 

In the meantime, you can just override the [`JAVA_OPTIONS`](https://github.com/DependencyTrack/dependency-track/blob/master/src/main/docker/Dockerfile#L19) environment variable like this:

```
environment:
  - "JAVA_OPTIONS=-XX:+UseParallelGC -Xmx6144m"
```

Note that the container will already use up to 90% of the memory that is available to it (`-XX:MaxRAMPercentage=90.0`). Using `-Xmx` will override this. See https://dzone.com/articles/best-practices-java-memory-arguments-for-container

**Reply by marcoamujica** [reply]
Thank you so much, @nscuro!  Very helpful, yet again.
